core/src/test/scala/zio/config/PropertyTypeTest.scala

```diff
@@ -0,0 +1,271 @@
+          propType = UriType,
+          genValid = genValidUriString,
+          parse = new URI(_)
+        ),
+        propertyTypeRoundtripSuite(
+          typeInfo = "Duration",
+          propType = DurationType,
+          genValid = Gen.sized(helpers.genDuration),
+          parse = Duration(_)
+        )
+      )
+    )
+
+object PropertyTypeTestUtils {
+  def propertyTypeRoundtripSuite[A](
+    typeInfo: String,
+    propType: PropertyType[String, A],
+    genValid: Gen[Random with Sized, String],
+    parse: String => A
+  ): Spec[TestEnvironment, TestFailure[Nothing], String, TestSuccess[Unit]] =
+    suite(s"${typeInfo}Type")(
+      testM(s"valid ${typeInfo} string roundtrip") {
+        check(genValid.map(_.toString))(assertValidRoundtrip(propType, parse))
+      },
+      testM(s"invalid ${typeInfo} string roundtrip") {
+        val invalidString = Gen.anyString.filter(s => Try(parse(s)).isFailure)
+        check(invalidString)(
+          assertInvalidRoundtrip(
+            propType,
+            s => PropertyReadError(s, typeInfo.toLowerCase)
+          )
+        )
+      }
+    )
+
+  def assertValidRoundtrip[A](
+    propType: PropertyType[String, A],
+    parse: String => A
+  )(s: String): TestResult =
+    assert(roundTrip(propType, s).map(parse), isRight(equalTo(parse(s))))
+
+  def assertInvalidRoundtrip[A](
+    propType: PropertyType[String, A],
+    propReadError: String => PropertyReadError[String]
+  )(s: String): TestResult =
+    assert(
+      roundTrip(propType, s),
+      isLeft(equalTo(propReadError(s)))
+    )
+
+  private def roundTrip[A](propType: PropertyType[String, A], s: String) =
+    propType.read(s).map(propType.write)
+
+  /** Generates all case permutations of a string */
+  private def casePermutations(s: String): List[String] = {
+    val max          = 1 << s.length
+    val s_           = s.toLowerCase
+    val combinations = Array.ofDim[String](max)
+    for (i <- 0 until max) {
+      val combination: Array[Char] = s_.toCharArray
+      for (j <- 0 until s_.length) {
+        if (((i >> j) & 1) == 1)
+          combination(j) = (combination(j) - 32).toChar
+      }
+      combinations(i) = combination.mkString
+    }
+    combinations.toList
+  }
+
+  import Gen._
+
+  val validBooleanStrings: List[String] =
+    casePermutations("true") ++ casePermutations("false")
+
+  val genValidBooleanStrings: Gen[Any, String] = Gen.fromIterable(validBooleanStrings)
+
+  private def genOptionalStr(gen: Gen[Random with Sized, String]) =
+    oneOf(const(""), gen)
+
+  private val genDigit0To9: Gen[Random, String] = char('0', '9').map(_.toString)
+  private val genDigit1To9: Gen[Random, String] = char('1', '9').map(_.toString)
+
+  /**
+   * Generates a string of up to n digits. Default maxLength digits is 100.
+   */
+  private def genDigits(maxLength: Int = 100): Gen[Random with Sized, String] =
+    sized(n => listOfN(math.min(n, maxLength))(genDigit0To9).map(_.mkString))
+

```

Comment:
@jdegoes Limiting digits to 100 by default. I went and checked the max and min numbers for this and both `BigInt` and `BigDecimal` and both are fine with them.